httpx/client.py

```diff
@@ -277,13 +278,21 @@ def _get_auth_middleware(
             )
 
         if trust_env:
-            netrc_login = get_netrc_login(request.url.authority)
-            if netrc_login:
-                username, _, password = netrc_login
-                return BasicAuthMiddleware(username=username, password=password)
+            netrc_info = self._get_netrc()
+            if netrc_info:
+                netrc_login = netrc_info.authenticators(request.url.authority)
+                if netrc_login:
+                    username, _, password = netrc_login
+                    return BasicAuthMiddleware(  # type: ignore
+                        username=username, password=password
+                    )
```

Comment:
I *think* the check stage in CI is failing because `.authenticators()` actually returns `[str, str, Optional[str]]` (i.e. the `password` is not guaranteed to be non-`None`).

The `# type: ignore` comment is probably misplaced, but this can also be fixed with an assert:

```python
username, _, password = netrc_login
assert password is not None
return BasicAuthMiddleware(username=username, password=password)
```